litgpt/chat/base.py

```diff
@@ -33,7 +34,12 @@ def generate(
         prompt: Tensor of shape (T) with indices of the prompt sequence.
         max_returned_tokens: The maximum number of tokens to return (given plus generated).
         temperature: Scales the predicted logits by 1 / temperature
-        top_k: If specified, only sample among the tokens with the k highest probabilities
+        top_k: If specified, only sample among the tokens with the k highest probabilities.
+        top_p: The cumulative probability threshold to consider in the sampling process.
```

Comment:
It's clear to us, but I wonder if it would help newcomers to explicitly state that this should be a value between 0 and 1?

Reply:
Something like this:

```python
"""
top_p: If specified, it represents the cumulative probability threshold to consider in the sampling process.
    In top-p sampling the next token is sampled from the highest probability tokens
    whose cumulative probability exceeds the threshold `top_p`. When specified
    it must be `0 <= top_p <= 1`, with `top_p=0` which is equivalent
    to sample the most probable token, while `top_p=1` samples from the whole distribution.
    It can be used in conjunction with `top_k` and `temperature` with the following order
    of application:

    1. `top_k`
    2. `temperature` scaling
    3. `top_p`

    For more details, see https://arxiv.org/abs/1904.09751
    or https://huyenchip.com/2024/01/16/sampling.html#top_p
"""
```

Reply:
Thanks, that sounds good to me. It's a bit long but I don't mind long if it's clear and helps users understand what's happening. I just have a small rewording but otherwise I like it! 

```
"""
top_p: If specified, it represents the cumulative probability threshold to consider in the sampling process.
    In top-p sampling, the next token is sampled from the highest probability tokens
    whose cumulative probability exceeds the threshold `top_p`. When specified,
    it must be `0 <= top_p <= 1`. Here, `top_p=0` is equivalent
    to sampling the most probable token, while `top_p=1` samples from the whole distribution.
    It can be used in conjunction with `top_k` and `temperature` with the following order
    of application:

    1. `top_k` sampling
    2. `temperature` scaling
    3. `top_p` sampling

    For more details, see https://arxiv.org/abs/1904.09751
    or https://huyenchip.com/2024/01/16/sampling.html#top_p
"""
```